scripts/load_test_ntc_server.js

```diff
@@ -0,0 +1,255 @@
+
+		// TODO: it's annoying that the load tester need to do so much parsing and serialization, can we do something about it?
+		// e.g. parse all frames at bootstrap? so runtime work is only to encode?
+
+		const binary_frame = this.game.frames.slice(
+			this.idx,
+			this.idx + this.game.frame_size
+		);
+		/*
+        console.log(binary_frame);
+        console.log(this.idx);
+        console.log(binary_frame[0].toString(2));
+        console.log(binary_frame[0] >> 5);
+        /**/
+
+		const frame_pojo = BinaryFrame.parse(binary_frame);
+
+		frame_pojo.gameid = this.gameid;
+		frame_pojo.ctime = Date.now() - this.global_local_ctime_start;
+
+		this.ws.send(BinaryFrame.encode(frame_pojo));
+
+		this.idx += this.game.frame_size;
+
+		// console.log(this.idx, this.game.frame_size, this.game.frames.length);
+
+		if (this.idx >= this.game.frames.length) {
+			// TODO: find a way to kill the test!
+			setTimeout(this.nextGame, 5000);
+		} else {
+			// schedule next frame
+			// TODO: Add speed multiplier (or should we not? hmmm)
+			const next_frame = this.game.frames.slice(
+				this.idx,
+				this.idx + this.game.frame_size
+			);
+			const next_frame_ctime = BinaryFrame.getCTime(next_frame);
+			const next_frame_ctime_diff = next_frame_ctime - this.game_start_ctime;
+			const next_frame_local_time =
+				this.game_start_local + next_frame_ctime_diff;
+
+			setTimeout(this.sendFrame, next_frame_local_time - Date.now());
+		}
+	}
+
+	nextGame() {
+		console.log(`${this.identifier} nextGame()`);
+
+		this.game = this.games.shift();
+		this.games.push(this.game);
+
+		++this.gameid;
+		this.idx = 0;
+
+		const first_frame = this.game.frames.slice(0, this.game.frame_size);
+		this.game_start_ctime = BinaryFrame.getCTime(first_frame);
+		this.game_start_local = Date.now();
+
+		try {
+			this.sendFrame();
+		} catch (err) {
+			console.log('==========');
+			console.log(err);
+			console.log(this.game.filename);
+			console.log(this.game.version);
+			console.log(this.game.frame_size);
+			console.log(first_frame);
+			return;
+		}
+	}
+}
+
+async function runWorker() {
+	// load the game files to be reused by all the players
+	for (const filename of localGameFiles) {
+		// super heavy cost at the start to do everything synchronously... but who cares... -_-
+		const buffer = new Uint8Array(
+			zlib.unzipSync(fs.readFileSync(new URL(`./${filename}`, import.meta.url)))
+		);
+		const version = buffer[0] >> 5;
+		const frame_size = BinaryFrame.FRAME_SIZE_BY_VERSION[version];
+
+		// console.log(buffer[0].toString(2).padStart(8, '0'));
+		// console.log((buffer[0] >> 5));
+
+		localGames.push({
+			frames: buffer,
+			filename,
+			version,
+			frame_size,
+		});
+
+		console.log(filename, buffer.length, buffer.length / frame_size);
+	}
+
+	const server_domain = process.env.server_domain;
+	const host_num = parseInt(process.env.host_num, 10);
+	// const { host_num, server_domain } = workerData;
+
+	// add 6 players to make 8 in the room
```

Comment:
Incorrect comment: basically this drops all users and create 8 fresh ones in the room